src/restLink.ts

```diff
@@ -1311,9 +1308,9 @@ export class RestLink extends ApolloLink {
   }
 
   public request(
-    operation: Operation,
-    forward?: NextLink,
-  ): Observable<FetchResult> | null {
+    operation: ApolloLink.Operation,
+    forward?: ApolloLink.ForwardFunction,
```

Comment:
```suggestion
    forward: ApolloLink.ForwardFunction,
```

As of v4, Apollo Client guarantees that a `forward` function is provided to `request`, so you don't need to make this optional. That should allow you to simplify a bit since you can remove some checks for the presence of this function.